pkg/api/mimo.go

```diff
@@ -0,0 +1,37 @@
+package api
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+type MaintenanceManifestState string
+
+const (
+	MaintenanceManifestStatePending    MaintenanceManifestState = "Pending"
+	MaintenanceManifestStateInProgress MaintenanceManifestState = "InProgress"
+	MaintenanceManifestStateCompleted  MaintenanceManifestState = "Completed"
+	MaintenanceManifestStateFailed     MaintenanceManifestState = "Failed"
+	MaintenanceManifestStateTimedOut   MaintenanceManifestState = "TimedOut"
+)
+
+type MaintenanceSet struct {
+	MissingFields
+
+	Name string `json:"name,omitempty"`
+}
+
+// MaintenanceManifest represents an instance of a MaintenanceSet running on a
+// given cluster.
+type MaintenanceManifest struct {
+	MissingFields
+
+	State      MaintenanceManifestState `json:"state,omitempty"`
```

Comment:
state transition can be painful, we may need to sort out the logic here, and adding some validation funcs to prevent unexpected transitions, e.g., from `Failed` to `Completed`?

Reply:
I added one of these in the documentation which hopefully clears things up.

I'm also not sure what we should do about finite state machine stuff without actually going and making it a proper finite state machine, but, considering the query which lists the docs filters by Pending, I don't think this will be able to come up.